infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java

```diff
@@ -158,9 +162,19 @@ private Container newContainer(
       List<EnvVar> envVars,
       String image,
       @Nullable String brokerVolumeName) {
+    // There's a chance the full image reference may be over the name limit of 63 chars
+    // so we need to remove registry hostname
+    Matcher matcher = IMAGE_PATTERN.matcher(image);
+    String containerName;
+    if (matcher.matches()) {
+      containerName = String.format("%s/%s", matcher.group("org"), matcher.group("image"));
+    } else {
+      containerName = image;
```

Comment:
will it be possible to extract this into method and cover with unit test + doc (the logic is not transparent IMO)

Reply:
I've extracted the method (and simplified it to strip everything but image name) and added tests.